src/SFA.DAS.EAS.Employer_Financial.Database/StoredProcedures/GetAllTransactionDetailsForAccountByDate.sql

```diff
@@ -70,51 +123,45 @@ INSERT INTO #output
 
-FROM [employer_financial].[Payment] p
-  INNER JOIN [employer_financial].[PaymentTransactionTypes] ptt ON ptt.TransactionType =  p.TransactionType
-  INNER JOIN [employer_financial].[PaymentMetaData] meta ON p.PaymentMetaDataId = meta.Id
-  INNER JOIN (SELECT PeriodEnd,AccountId,ukprn, EmpRef, TransactionDate, DateCreated, LevyDeclared, EnglishFraction FROM employer_financial.TransactionLine WHERE DateCreated >= @FromDate AND 
-        DateCreated <= @ToDate) transLine ON transLine.AccountId = p.AccountId AND transLine.PeriodEnd = p.PeriodEnd AND transLine.Ukprn = p.Ukprn
-  LEFT JOIN [employer_financial].[Payment] pays1 
-	ON pays1.AccountId = p.AccountId 
-		AND pays1.Ukprn = p.Ukprn 
-		AND pays1.FundingSource = 1 
-		AND pays1.PaymentMetaDataId = meta.id
-  LEFT JOIN [employer_financial].[Payment] pays2 
-	ON pays2.AccountId = p.AccountId 
-	AND pays2.Ukprn = p.Ukprn 
-	AND pays2.FundingSource = 2 
-	AND pays2.PaymentMetaDataId = meta.id
-  LEFT JOIN [employer_financial].[Payment] pays3 
-	ON pays3.AccountId = p.AccountId 
-	AND pays3.Ukprn = p.Ukprn 
-	AND pays3.FundingSource = 3 
-	AND pays3.PaymentMetaDataId = meta.id
-  WHERE 
-	  p.AccountId = @accountid AND
-	  p.FundingSource IN (1,2,3)  
-  GROUP BY p.AccountId, 
-	  p.Ukprn, 
-	  p.TransactionType, 
-	  ptt.[Description],
-	  p.Uln, 
-	  meta.ProviderName, 
-	  meta.ApprenticeshipCourseName, 
-	  meta.ApprenticeshipCourseLevel, 
-	  meta.PathwayName, 
-	  meta.ApprenticeName
+-- Join the Views together
+SELECT DISTINCT 
+	COALESCE(MAX(accountFunds1.DateCreated),MAX(accountFunds2.DateCreated),MAX(accountFunds3.DateCreated)) AS DateCreated,
+	COALESCE(MAX(accountFunds1.TransactionTypeDesc),MAX(accountFunds2.TransactionTypeDesc),MAX(accountFunds3.TransactionTypeDesc)) AS [Description],
+	COALESCE(MAX(accountFunds1.TrainingProvider),MAX(accountFunds2.TrainingProvider),MAX(accountFunds3.TrainingProvider)) AS TrainingProvider,
+	uniquePayments.Uln AS Uln, 
+	COALESCE(MAX(accountFunds1.Apprentice),MAX(accountFunds2.Apprentice),MAX(accountFunds3.Apprentice)) AS Apprentice,
+	COALESCE(MAX(accountFunds1.ApprenticeTrainingCourse), MAX(accountFunds2.ApprenticeTrainingCourse), MAX(accountFunds3.ApprenticeTrainingCourse)) AS ApprenticeTrainingCourse,
+	COALESCE(MAX(accountFunds1.ApprenticeTrainingCourseLevel), MAX(accountFunds2.ApprenticeTrainingCourseLevel), MAX(accountFunds3.ApprenticeTrainingCourseLevel)) AS ApprenticeTrainingCourseLevel,
+	COALESCE((SUM(accountFunds1.[Amount]) * -1), 0) AS PaidFromLevy,
+	COALESCE((SUM(accountFunds3.Amount) * -1), 0) AS EmployerContribution,
+	COALESCE((SUM(accountFunds2.Amount) * -1), 0) AS GovermentContribution,
+	COALESCE((SUM(accountFunds1.[Amount]) * -1),0) + COALESCE((SUM(accountFunds2.[Amount]) * -1),0) + COALESCE((SUM(accountFunds3.[Amount]) * -1),0) AS Total
+FROM UniqueApprenticePaymentRecords uniquePayments
+LEFT JOIN AllFunds accountFunds1 
```

Comment:
I think the suffix 1/2/3 == levy/employer/government - wondering if using descriptive names would make the intent easier to work out.

Reply:
Agreed and done